samtranslator/swagger/swagger.py

```diff
@@ -620,15 +625,15 @@ def set_path_default_apikey_required(self, path):
         :param string path: Path name
         """
 
+        valid_non_http_method_sections = ["parameters", "summary", "description", "$ref", "servers"]
         for method_name, _ in self.get_path(path).items():
-            # Excluding parameters section
-            if method_name == "parameters":
+            # Excluding non-http-method sections https://swagger.io/specification/
+            if method_name in valid_non_http_method_sections:
                 continue
 
             normalized_method_name = self._normalize_method_name(method_name)
             # It is possible that the method could have two definitions in a Fn::If block.
             for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
```

Comment:
We should use `method_name` instead of `normalized_method_name` here and at line 539.
If a customer defines the method as `Get` for example, we are trying to get the normalized version `get`of this and this will fail with a `KeyError` exception.

Reply:
There is a separate PR for that: https://github.com/aws/serverless-application-model/pull/2031/files